File: app/src/main/java/com/bsunk/hapanel/ui/main/MainActivityContract.java
package com.bsunk.hapanel.ui.main;

/**
 * Created by Bharat on 3/8/2017.
 */

public interface MainActivityContract {

    interface View {

        void startStopConnectionService(boolean shouldStart);
        void setTitle(String name);
        void setConnectionImage(int event);
        void keepScreenOn(boolean shouldKeepOn);

    }

    interface Presenter {

        void subscribe(MainActivityContract.View view);
        void unSubscribe();
        void setView(MainActivityContract.View view);
        void initToolbarTitle();

    }
}

File: app/src/main/java/com/bsunk/hapanel/ui/main/MainActivityContract.kt
package com.bsunk.hapanel.ui.main

/**
 * Created by Bharat on 3/8/2017.
 */

interface MainActivityContract {

    interface View {

        fun startStopConnectionService(shouldStart: Boolean)
        fun setTitle(name: String)
        fun setConnectionImage(event: Int)
        fun keepScreenOn(shouldKeepOn: Boolean)

    }

    interface Presenter {

        fun subscribe(view: MainActivityContract.View)
        fun unSubscribe()
        fun setView(view: MainActivityContract.View)
        fun initToolbarTitle()

    }
}
